### PR TITLE
Warn about using --rpc-external and --ws-external options

### LINKED ISSUE #4445: Print warnings when RPC are exposed publicly
It is not safe to expose RPC publicly without a proxy server to block some configuration RPC endpoints such as `system_addReservedPeer`.

When `--http-external` or `--ws-external` flag is passed, the node should print some warning message to inform user this is could be dangerous. Maybe add `--unsafe-http-external` / `--unsafe-ws-external` that suppress the warning.

The help message for `--*-external` flag should also mention they are unsafe.

Could even panic and require use of `--unsafe-*-external` when `--validator` flag is also used.

More documentations are also required (Out of scope of this issue, just for reference):

- Which endpoints are safe to expose and which are not
- How to setup a proxy to block / require auth for unsafe RPC endpoints

cc @tomusdrw

### PR DESCRIPTION
**Pull request by DarkEld3r**
Closes #4445.

### CODE REVIEW

File: client/cli/src/lib.rs [hunk 1]
```diff
@@ -931,6 +931,28 @@ where
...
 }
 
+fn interface_str(
+	is_external: bool,
+	is_unsafe_external: bool,
+	is_validator: bool
+) -> Result<&'static str, error::Error> {
+	if is_external && is_validator {
+		return Err(error::Error::Input("--rpc-external and --ws-external options shouldn't be \
+		used if the node is running as vaidator.".to_owned()));
```
Comment (tomusdrw) [inline client/cli/src/lib.rs]:
```suggestion
		used if the node is running as validator.".to_owned()));
```

File: client/cli/src/lib.rs [hunk 2]
```diff
@@ -931,6 +931,28 @@ where
 	Ok(config)
 }
 
+fn interface_str(
+	is_external: bool,
+	is_unsafe_external: bool,
+	is_validator: bool
+) -> Result<&'static str, error::Error> {
+	if is_external && is_validator {
```
Comment (tomusdrw) [inline client/cli/src/lib.rs]:
```suggestion
	if is_validator && !is_unsafe_external {
```

When you run a dev chain you probably don't care about the risk at all, it should be allowed to use `unsafe` flag to override this.

File: client/cli/src/lib.rs [hunk 3]
```diff
@@ -931,6 +931,28 @@ where
...
+		return Err(error::Error::Input("--rpc-external and --ws-external options shouldn't be \
+		used if the node is running as vaidator.".to_owned()));
+	}
+
+	if is_external {
+		log::warn!("It isn't safe to expose RPC publicly without a proxy server. \
```
Comment (tomusdrw) [inline client/cli/src/lib.rs]:
```suggestion
		log::warn!("It isn't safe to expose RPC publicly without a proxy server that filters available set of RPC methods. \
```

File: client/cli/src/lib.rs [hunk 4]
```diff
@@ -931,6 +931,28 @@ where
...
+		Use `--unsafe-rpc-external` to suppress the warning if you understand the risks.");
```
Comment (tomusdrw) [inline client/cli/src/lib.rs]: To be fair I wouldn't give an option to suppress the warning, I think `unsafe` should just be used to be able to run the node even if you are a validator. I'd keep the warning whenever you run with `is_external || is_unsafe_external`.

If you agree we should move this message to the validator error instead and update the condition above.

File: client/cli/src/lib.rs (hunk 1 above)
Comment (HCastano) [inline client/cli/src/lib.rs]:
```suggestion
		used if the node is running as validator.".to_owned()));
```

File: client/cli/src/params.rs [hunk 5]
```diff
@@ -438,16 +438,30 @@ pub struct RunCmd {
...
+	/// Listen to all RPC interfaces.
+	///
+	/// Same as `--rpc-external`.
+	#[structopt(long = "unsafe-rpc-external")]
+	pub unsafe_rpc_external: bool,
+
 	/// Listen to all Websocket interfaces.
 	///
-	/// Default is local.
+	/// Default is local. Note: it isn't safe to expose RPC publicly without a proxy server.
```
Comment (tomusdrw) [inline client/cli/src/params.rs]:
```suggestion
	/// Default is local. Note: not all RPC methods are safe to be exposed publicly. Use a RPC proxy server to filter out dangerous methods. More details: (link to wiki).
```

Does anyone know if we have node configuration documentation somewhere? CC @shawntabrizi ?

File: client/cli/src/lib.rs [hunk 6]
```diff
@@ -931,6 +931,28 @@ where
 	Ok(config)
 }
 
+fn interface_str(
+	is_external: bool,
+	is_unsafe_external: bool,
+	is_validator: bool
```
Comment (tomusdrw) [inline client/cli/src/lib.rs]:
```suggestion
	is_validator: bool,
```

**Comment by DarkEld3r** [comment]
In progress till I figure which link to the documentation should be used.

File: client/cli/src/params.rs (hunk 5 above)
Comment (DarkEld3r) [reply client/cli/src/params.rs]: Unfortunately, it looks like we don't have such documentation.

**Pull request merged**